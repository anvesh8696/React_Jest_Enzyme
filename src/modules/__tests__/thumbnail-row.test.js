import React from 'react';
import ThumbnailRow from '../thumbnail-row'
import Thumbnail from '../thumbnail';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import fakePhotos from '../__data__/photos.fake.js';

describe("<ThumbnailRow />", () => {
	let wrapper, instance;
	let cols = 3;
	let rowIndex = 1;

	beforeEach(() => {
		const photosSubSet = fakePhotos.slice(0, cols);
		wrapper = shallow(
			<ThumbnailRow
				rowIndex={rowIndex}
				cols={cols}
				photos={photosSubSet}
			/>
		);
		instance = wrapper.instance();
	});

	it("populated thumbnails on page", () => {
		expect(wrapper.find(Thumbnail)).toHaveLength(cols);
	});

	it('assigns the thumbnails with correct column index', () => {
		let secondThumb = wrapper.find(Thumbnail).at(1);
		expect(secondThumb.props().colIndex).toBe(1);
	});

	it('state was populated with row index', () => {
		expect(instance.props.rowIndex).toEqual(rowIndex);
	});

	it('assigns the thumbnails the correct photo', () => {
		let secondThumb = wrapper.find(Thumbnail).at(1);
		expect(secondThumb.props().photo).toEqual(fakePhotos[1]);
	});

	it('matches snaptshot', () => {
		expect(toJson(wrapper)).toMatchSnapshot();
	});


});