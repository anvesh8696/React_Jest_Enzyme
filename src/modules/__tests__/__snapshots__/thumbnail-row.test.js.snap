exports[`<ThumbnailRow /> matches snaptshot 1`] = `
<div
  className="thumbnailRow">
  <Thumbnail
    colIndex={0}
    photo={
      Object {
        "id": 0,
        "thumbnailUrl": "thumbnail0",
        "url": "fakePhoto0",
      }
    }
    rowIndex={1} />
  <Thumbnail
    colIndex={1}
    photo={
      Object {
        "id": 1,
        "thumbnailUrl": "thumbnail1",
        "url": "fakePhoto1",
      }
    }
    rowIndex={1} />
  <Thumbnail
    colIndex={2}
    photo={
      Object {
        "id": 2,
        "thumbnailUrl": "thumbnail2",
        "url": "fakePhoto2",
      }
    }
    rowIndex={1} />
</div>
`;
